clang/lib/Sema/SemaCoroutine.cpp

```diff
@@ -694,6 +707,9 @@ bool Sema::ActOnCoroutineBodyStart(Scope *SC, SourceLocation KWLoc,
   auto *ScopeInfo = getCurFunction();
   assert(ScopeInfo->CoroutinePromise);
 
+  if (ScopeInfo->FirstCoroutineStmtLoc == KWLoc)
```

Comment:
NIT: add a comment explaning why we don't do this every time:

```cpp
// Avoid duplicate errors, report only on first keyword.
```

Reply:
Done.